feat: Release 1.17 (#166)
cdr/docs: changelog/1.17.0.md

---
title: "1.17.0"
description: "Released on 03/17/2021"
---

### Breaking Changes ❗

> - (?) Wsp orgs migration

Author: Might need @f0ssel 's help here

### Features ✨

> - Environment Providers **TODO**

Author: @khorne3 I kind of left this one open for you and @f0ssel to do your thing on. Sorry about that, just wasn't sure to what extent we'll list out the changes

- Autostart: Autostart allows you to set the time when Coder
  automatically starts and builds your environments.
  - web: A new onboarding wizard is available for setting a preferred Autostart
    time.
  - web: Visit your account preferences > Autostart to set a time and any
  environments that should be automatically started. We recommend enabling this
  feature for the environments you use the most.
  - web: Autostart action has been added to the Audit Log
  - web: A new checkbox for opting into Autostart is available on the Create and

>     Edit Environment modals.

Author: @khorne3 Not sure if best format or not, but it felt like a nice way to present the feature and list all the places it's found.

cc @fuskovic for anything major missed

- web: Environment stop actions have been added to the Audit Log
- web: Show most recent Build Log for offline environments
- cli: Improved sign in flow with (cdr/coder-cli#274)
- web: Improved error messages within the Build Log
  - Improve messages when '/coder/configure' is not executable
- code-web: Upgrade to code-server version 3.9.0
- jetbrains: The Mac OS keymap plugin is now automatically installed
- web: Resources section on Environments page has been redesigned
- web: Apply consistent design theme across all Preferences pages

### Bug Fixes 🐛

- web: Fix DevURL error page links using incorrect URIs.
- infra: Fix timeout when deleting users with many environments
- infra: Increase offline timeout in trial versions of Coder
- web: Use proper casing when referencing 'GitHub' and 'GitLab'
- infra: Fix renaming an OAuth provider without creating a duplicate. Migrating

>   to 1.17.0 will remove these duplicates. They are not visible in the UI.

Author: cc @Emyrk to verify

Reviewer: The language feels odd here, but it is correct.
> Fix renaming an OAuth provider without creating a duplicate.

Maybe??

Fix renaming an OAuth provider to correctly rename the existing OAuth provider without creating a duplicate.

- infra: Correct environment statuses during build and shutdown states
- web: Fix SiteAdmin users receiving a notification that their SSH keys are out
  of sync after creating an initial environment.
  - Note: We do not recommend using the SiteAdmin user beyond initial setup

### Security Updates 🔐

- Resetting the SiteAdmin password will now always be marked as temporary and

>   require a new one to be set upon sign in.

Author: cc @jawnsy to verify

Reviewer 2: this is true for `cemanager reset-admin-password`, but the changelog entry is ambiguous here (i.e. if you reset via the UI or API then this is not true)

Author: Agreed, I wasn't sure on scope. Some users `kube exec` that command I believe - do you know?

Reviewer 2: sorry I phrased my comment badly, suggest we use this in the changelog instead (I'd have done this as a suggestion but I still have no idea how to mark those up correctly):

```
- When resetting the SiteAdmin password using the
  `cemanager reset-admin-password` tool, Coder now requires a new
  password to be set upon next login (it is marked as a temporary
  password).
```

Author: That's awesome ty !

Reviewer 2: The parenthetical part is optional, depending o whether we want to talk about the implementation detail of the temporary password flag

Author: https://github.com/cdr/docs/pull/166/commits/57f04f959dcd596a078f810e3e89fab915586d20

---
title: "1.17.0"
description: "Released on 03/17/2021"
---

### Breaking Changes ❗

- helm: The `namespaceWhitelist` field on the Coder helm charts [has been
  removed and the organization namespaces feature has been

>   deprecated](../admin/access-control/organizations.md#namespaces).

Reviewer 3: Should add:

- Deployments that had this value set cannot add namespaces, and cannot remove namespaces unless there are no environments still using them
- Existing environments in whitelisted namespaces will continue to work
- Environments cannot be created in whitelisted namespaces anymore

### Features ✨

- [Workspace Providers](../admin/workspace-providers/index.md): Workspace
  providers enable a single Coder deployment to provision and manage workspaces
  across multiple Kubernetes clusters and namespaces.
  - web: New configuration page available via **Manage** > **Admin** >
    **Workspace Providers**
  - infra: New support for the use of multiple Kubernetes clusters and
    namespaces
- [Autostart](../environments/autostart.md): Autostart allows you to set the
  time when Coder automatically starts and builds your environments.
  - web: New onboarding wizard available for setting a preferred Autostart time.
  - web: New configuration page available via **Account Preferences** >
    **Autostart** that allows you to set a time and choose the environments that
    should be automatically started. We recommend enabling this feature for the
    environments you use the most.
  - web: New checkbox for opting into Autostart is available on the Create and
    Edit Environment modals.
  - web: Autostart action has been added to the Audit Log

- web: Environment stop actions have been added to the Audit Log
- web: UI shows most recent Build Log for offline environments

> - cli: Improved sign-in flow with (cdr/coder-cli#274)

Reviewer 3: with what?

Author: ```suggestion
- cli: Improved sign-in flow
```

---
title: "1.17.0"
description: "Released on 03/17/2021"
---

### Breaking Changes ❗

- helm: The `namespaceWhitelist` field on the Coder helm charts [has been
  removed and the organization namespaces feature has been
  deprecated](../admin/access-control/organizations.md#namespaces).
  - Deployments that had this value set cannot add namespaces, and cannot remove
    namespaces unless there are no environments still using them

> - Existing environments in whitelisted namespaces will continue to work
> - Environments cannot be created in whitelisted namespaces anymore

Author: I believe these should be sub-points

```suggestion
  - Existing environments in whitelisted namespaces will continue to work
  - Environments cannot be created in whitelisted namespaces anymore
```

### Features ✨

- [Workspace Providers](../admin/workspace-providers/index.md): Workspace
  providers enable a single Coder deployment to provision and manage workspaces
  across multiple Kubernetes clusters and namespaces.
  - web: New configuration page available via **Manage** > **Admin** >
    **Workspace Providers**
  - infra: New support for the use of multiple Kubernetes clusters and

>     namespaces

Author: Think we should also add a `web:` point for the new field in Create/Edit environment dialog

---
title: "1.17.0"
description: "Released on 03/17/2021"
---

### Breaking Changes ❗

- helm: The `namespaceWhitelist` field on the Coder helm charts [has been
  removed and the organization namespaces feature has been

>   deprecated](../admin/access-control/organizations.md#namespaces).

Author: **Important**

IMO we shouldn't link to documentation within the changelog, unless the link contains a version. The changelog
should be linked by any other documentation, but since the changelog is immutable, it shouldn't contain links to
pages that may change across versions or be removed.

Reviewer 4: That's a good point -- will remove the links

Author: After that, I actually think we're looking ready! Thanks so much :taco:

  - Deployments that had this value set cannot add namespaces, and cannot remove
    namespaces unless there are no environments still using them
  - Existing environments in whitelisted namespaces will continue to work

>   - Environments cannot be created in whitelisted namespaces anymore

Reviewer 3: ```suggestion
  - Environments cannot be created in whitelisted namespaces any longer
```

### Features ✨

> - [Workspace Providers](../admin/workspace-providers/index.md): Workspace

Author: **Important** - refer to note on changelog containing links to volatile documents

  providers enable a single Coder deployment to provision and manage workspaces
  across multiple Kubernetes clusters and namespaces.
  - web: New configuration page available via **Manage** > **Admin** >
    **Workspace Providers**
  - web: New option to choose your workspace provider in the Create and Edit
    Environment modals
  - infra: New support for the use of multiple Kubernetes clusters and
    namespaces

> - [Autostart](../environments/autostart.md): Autostart allows you to set the

Author: **Important** - refer to note on changelog containing links to volatile documents

---
title: "1.17.0"
description: "Released on 03/17/2021"
---

### Breaking Changes ❗

- helm: The `namespaceWhitelist` field on the Coder helm charts has been
  removed and the organization namespaces feature has been
  deprecated.
  - Deployments that had this value set cannot add namespaces, and cannot remove
    namespaces unless there are no environments still using them
  - Existing environments in whitelisted namespaces will continue to work

>   - Environments cannot be created in whitelisted namespaces any longer

Reviewer 3: ```suggestion
  - Environments cannot be created in whitelisted namespaces any longer
  - Organizations can't be removed from a workspace provider (see below) until all environments in that combination of organization and workspace provider are deleted. Users in an organization will be able to create environments in the built-in workspace provider until all environments in the built-in provider are deleted and the organization is removed from the whitelist.
```

---
title: "1.17.0"
description: "Released on 03/17/2021"
---

### Breaking Changes ❗

- helm: The `namespaceWhitelist` field on the Coder helm charts has been
  removed and the organization namespaces feature has been
  deprecated.
  - Deployments that had this value set cannot add namespaces, and cannot remove
    namespaces unless there are no environments still using them
  - Existing environments in whitelisted namespaces will continue to work
  - Environments cannot be created in whitelisted namespaces any longer
  - Organizations can't be removed from a workspace provider (see below) until all

>   environments in that organization *and* workspace provider are deleted. Users
>   in an organization can create environments in the built-in workspace provider
>   until *all* environments in that provider are deleted and the organization is
>   removed from the whitelist.

Reviewer 3: ```suggestion
    environments in that organization *and* workspace provider are deleted. Users
    in an organization can create environments in the built-in workspace provider
    until *all* environments in that provider are deleted and the organization is
    removed from the whitelist.
```

    **Autostart** that allows you to set a time and choose the environments that
    should be automatically started. We recommend enabling this feature for the
    environments you use the most.
  - web: New checkbox for opting into Autostart is available on the Create and
    Edit Environment modals.
  - web: Autostart action has been added to the Audit Log

- web: Environment stop actions have been added to the Audit Log
- web: UI shows most recent Build Log for offline environments
- cli: Improved sign-in flow with (cdr/coder-cli#274)
- web: Improved error messages within the Build Log
  - Improved messages when '/coder/configure' is not executable
- code-web: Upgraded to code-server version 3.9.0
- jetbrains: The Mac OS keymap plugin is now automatically installed
- web: Resources section on Environments page has been redesigned
- web: Applied consistent design theme across all Preferences pages

### Bug Fixes 🐛

- web: Fixed DevURL error page links using incorrect URIs.
- infra: Fixed timeout when deleting users with many environments
- infra: Increased offline timeout in trial versions of Coder
- web: Fixed casing when referencing 'GitHub' and 'GitLab'

> - infra: Fixed issue renaming an OAuth provider without creating a duplicate.

Author: Maybe we can go the opposite way here:

```
- infra: Fixed an issue whereby renaming an OAuth provider caused a duplicate to be created
  Migrating to 1.17.0...
```

Reviewer: that works
